### PR TITLE
test: Shorten genai-perf test time, fail fast on server startup, and upgrade to 24.06

### PR DESCRIPTION
**Pull request by rmccorm4**
This PR makes the following changes:
1. Adds `--request-count 10` to `genai-perf` when running `triton profile` in the test scripts. This is done to reduce risk of github action jobs timing out, use less overall github action minutes, and speed up local development time when running tests locally.

2. Removes the `triton profile` call for the `http` subtest, since `genai-perf` is using `grpc` under the hood for both tests, it was just wasting test time without adding value.

3. Improves the checks around server readiness to detect if the server process failed to start altogether, rather than waiting the full timeout period only polling the endpoints.

Note that locally, `--request-count 1` and `--request-count 10` both took about the same time of ~4-5 seconds instead of the previous default behavior taking 1m30s - 2m or so.

Overall, this should reduce the e2e test time by about 3-5 minutes in the succesful case, and possibly even more when server startups fail for some reason.

---

Example successful tests with genai-perf speedup changes, full test time is ~50seconds:

```
root@ced35d0-lcedt:/mnt/triton/cli/triton_cli/tests# pytest -v
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.10.12, pytest-8.2.2, pluggy-1.5.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /mnt/triton/cli/triton_cli
configfile: pyproject.toml
plugins: timeout-2.3.1, mock-3.14.0
collected 55 items

test_cli.py::TestRepo::test_clear[None] PASSED                                                                                                                                          [  1%]
test_cli.py::TestRepo::test_clear[tmp/models] PASSED                                                                                                                                    [  3%]
test_cli.py::TestRepo::test_import_known_model[None-llama-2-7b] PASSED                                                                                                                  
...
test_cli.py::TestRepo::test_import_vllm[vllm-model-hf:gpt2] PASSED                                                                                                                      [ 63%]
test_cli.py::TestRepo::test_repo_add_trtllm_build[trtllm-model-hf:gpt2] SKIPPED (Only run for TRT-LLM image)                                                                            [ 65%]
test_cli.py::TestRepo::test_import_trtllm_prebuilt SKIPPED (Pre-built TRT-LLM engines not available)                                                                                    [ 67%]
test_cli.py::TestRepo::test_import_no_source PASSED                                                                                                                                     [ 69%]
test_cli.py::TestRepo::test_remove PASSED                                                                                                                                               [ 70%]
test_cli.py::TestRepo::test_remove_nonexistent PASSED                                                                                                                                   [ 72%]
test_cli.py::TestRepo::test_list[None] PASSED                                                                                                                                           [ 74%]
test_cli.py::TestRepo::test_list[tmp/models] PASSED                                                                                                                                     [ 76%]
test_cli.py::TestRepo::test_triton_profile PASSED                                                                                                                                       [ 78%]
test_cli.py::TestRepo::test_triton_metrics[mock_llm] PASSED                                                                                                                             [ 80%]
test_cli.py::TestRepo::test_triton_config[mock_llm] PASSED                                                                                                                              [ 81%]
test_cli.py::TestRepo::test_triton_status[mock_llm] PASSED                                                                                                                              [ 83%]
test_e2e.py::TestE2E::test_tensorrtllm_e2e[grpc] SKIPPED (Only run for TRT-LLM image)                                                                                                   [ 85%]
test_e2e.py::TestE2E::test_tensorrtllm_e2e[http] SKIPPED (Only run for TRT-LLM image)                                                                                                   [ 87%]
test_e2e.py::TestE2E::test_vllm_e2e[grpc] SKIPPED (Only run for VLLM image)                                                                                                             [ 89%]
test_e2e.py::TestE2E::test_vllm_e2e[http] SKIPPED (Only run for VLLM image)                                                                                                             [ 90%]
test_e2e.py::TestE2E::test_non_llm[grpc] PASSED                                                                                                                                         [ 92%]
test_e2e.py::TestE2E::test_non_llm[http] PASSED                                                                                                                                         [ 94%]
test_e2e.py::TestE2E::test_mock_llm[grpc] PASSED                                                                                                                                        [ 96%]
test_e2e.py::TestE2E::test_mock_llm[http] PASSED                                                                                                                                        [ 98%]
test_library.py::test_version PASSED                                                                                                                                                    [100%]

=============================================================================== 49 passed, 6 skipped in 50.46s ================================================================================
```

Example run with server startup failures, failing fast rather than waiting 60 seconds on each failed startup polling the endpoints:
```
================================================================================== short test summary info ===================================================================================
FAILED test_cli.py::TestRepo::test_triton_metrics[mock_llm] - Exception: 'triton start' PID 824213 was in a zombie state.
FAILED test_cli.py::TestRepo::test_triton_config[mock_llm] - Exception: 'triton start' PID 824227 was in a zombie state.
FAILED test_cli.py::TestRepo::test_triton_status[mock_llm] - Exception: 'triton start' PID 824242 was in a zombie state.
FAILED test_e2e.py::TestE2E::test_mock_llm[grpc] - Exception: 'triton start' PID 824254 was in a zombie state.
FAILED test_e2e.py::TestE2E::test_mock_llm[http] - Exception: 'triton start' PID 824266 was in a zombie state.
======================================================================== 5 failed, 50 deselected, 1 warning in 5.77s =========================================================================
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -50,16 +50,16 @@ dependencies = [
...
     "directory-tree == 0.0.4", # may remove in future
     "docker == 6.1.3",
-    "genai-perf @ git+https://github.com/triton-inference-server/client.git@r24.05#subdirectory=src/c++/perf_analyzer/genai-perf",
+    "genai-perf @ git+https://github.com/triton-inference-server/client.git@r24.06#subdirectory=src/c++/perf_analyzer/genai-perf",
     # TODO: rely on tritonclient to pull in protobuf and numpy dependencies?
     "numpy >=1.21,<2",
     "protobuf>=3.7.0",
     "prometheus-client == 0.19.0",
-    "psutil >= 5.9.5", # may remove later
+    "psutil < 6", # psutil 6.x may be causing some hangs
```
Comment (rmccorm4) [inline pyproject.toml:58]: revert: this appear to have no effect.

File: tests/utils.py [hunk 2]
```diff
@@ -123,21 +126,50 @@ def run_server(self, repo=None, mode="local"):
...
             print(
-                "Waiting for server to be ready ",
+                "[DEBUG] Waiting for server to be ready ",
                 round(timeout - (time.time() - start)),
                 flush=True,
             )
             time.sleep(1)
             try:
+                print(f"[DEBUG] Checking status of 'triton start' PID {self.pid}...")
+                self.check_pid()
```
Comment (rmccorm4) [inline tests/utils.py:165]: Should sanity check that it is indeed kill_server that is hanging prior to these changes that do additional checks/communication with the 'triton start' PID.

File: .github/workflows/python-package.yaml [hunk 3]
```diff
@@ -36,7 +36,7 @@ jobs:
   build:
     runs-on: ${{ matrix.os }}
     container:
-      image: nvcr.io/nvidia/tritonserver:24.05-py3
+      image: nvcr.io/nvidia/tritonserver:24.06-py3
```
Comment (rmccorm4) [inline .github/workflows/python-package.yaml:39]: revert: had no change to failing/hanging test issue

Comment (rmccorm4) [reply .github/workflows/python-package.yaml:39]: Will keep

File: tests/utils.py (hunk 2 above)
Comment (rmccorm4) [reply tests/utils.py:165]: Investigation will be a follow-up

File: src/triton_cli/client/client.py [hunk 4]
```diff
@@ -256,7 +256,7 @@ def callback(user_data, result, error):
         try:
             while completed_requests != num_requests:
                 result = user_data._completed_requests.get()
-                if type(result) == InferenceServerException:
+                if isinstance(result, InferenceServerException):
```
Comment (KrishnanPrash) [inline src/triton_cli/client/client.py:259]: Nice Change!

Comment (rmccorm4) [reply src/triton_cli/client/client.py:259]: `ruff` made me do it 👼

File: .github/workflows/python-package.yaml [hunk 5]
```diff
@@ -65,4 +65,4 @@ jobs:
         export CI_PIPELINE="GITHUB_ACTIONS"
         python3 -m pip install pytest pytest-timeout pytest-cov psutil
         cd tests
-        pytest -vv --doctest-modules --junitxml=junit/test-results.xml --cov=triton_cli --cov-report=xml --cov-report=html --ignore-glob=test_models
+        pytest -s -vv --doctest-modules --junitxml=junit/test-results.xml --cov=triton_cli --cov-report=xml --cov-report=html --ignore-glob=test_models
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply .github/workflows/python-package.yaml:68]: Figured I'd keep it around until we investigate/fix the hanging test issues later. Can add some debug info for now, we can remove later if we find it more harm than good.

**Review by KrishnanPrash — APPROVED** [review]
Great changes that reduce CI time.

**Pull request merged**